src/main/java/org/jabref/logic/protectedterms/ProtectedTermsParser.java

```diff
@@ -29,30 +30,37 @@ public class ProtectedTermsParser {
     private String location;
 
     public void readTermsFromResource(String resourceFileName, String descriptionString) {
-        try {
-            Path path = Path.of(Objects.requireNonNull(ProtectedTermsLoader.class.getResource(Objects.requireNonNull(resourceFileName))).toURI());
-            readTermsList(path);
-            description = descriptionString;
-            location = resourceFileName;
-        } catch (URISyntaxException e1) {
-            LOGGER.error("");
+        description = descriptionString;
+        location = resourceFileName;
+        try (InputStream inputStream = ProtectedTermsLoader.class.getResourceAsStream(Objects.requireNonNull(resourceFileName))) {
+            if (inputStream == null) {
+                LOGGER.error("Cannot find resource '{}' ({})", resourceFileName, descriptionString);
+                return;
+            }
+            readTermsList(inputStream);
+        } catch (IOException e) {
+            LOGGER.error("Cannot open resource '{}'", resourceFileName, e);
         }
     }
 
     public void readTermsFromFile(Path path) {
-        location = path.toAbsolutePath().toString();
-        readTermsList(path);
-    }
-
-    private void readTermsList(Path path) {
+        path = path.toAbsolutePath();
         if (!Files.exists(path)) {
             LOGGER.warn("Could not read terms from file {}", path);
             return;
         }
-        try (Stream<String> lines = Files.lines(path, StandardCharsets.UTF_8)) {
```

Comment:
Files lines does not work with graal? Tbh this looks like a major drawback here.

Reply:
BufferedReader#lines seems to be somewhat of an equivalent here...

Reply:
In line 60, I use exactly this function.

This call is only a wrapper for the call if a user supplies a file!

GraalVM does not support `.class.getResource(` with hacks. See https://github.com/oracle/graal/issues/7682 for a longer discussion. Especially https://github.com/oracle/graal/issues/7682#issuecomment-1786704111

> The reason NativeImageFileSystem behaves like JarFileSystem, not like UnixFileSystem, is that, unlike UnixFileSystem, you can't, for example, delete a file (resource) on both NativeImageFileSystem and JarFileSystem. The resources in NativeImageFileSystem are baked into the read-only section of the image heap